tests/integration/local/start_api/lambda_authorizers/test_swagger_authorizer_definitions.py

```diff
@@ -0,0 +1,332 @@
+
+
+@parameterized_class(
+    ("template_path", "endpoint"),
+    [
+        ("/testdata/start_api/lambda_authorizers/swagger-api.yaml", "requestauthorizerswaggertoken"),
+        ("/testdata/start_api/lambda_authorizers/swagger-api.yaml", "requestauthorizerswaggerrequest"),
+        ("/testdata/start_api/lambda_authorizers/swagger-api.yaml", "requestauthorizeropenapi"),
+        ("/testdata/start_api/lambda_authorizers/swagger-http.yaml", "requestauthorizer"),
+        ("/testdata/start_api/lambda_authorizers/swagger-http.yaml", "requestauthorizersimple"),
+    ],
+)
+class TestSwaggerLambdaAuthorizer500(StartApiIntegBaseClass):
+    parameter_overrides = {"AuthHandler": "app.throws_exception"}
+
+    def setUp(self):
+        self.url = f"http://127.0.0.1:{self.port}"
+
+    @pytest.mark.flaky(reruns=3)
+    @pytest.mark.timeout(timeout=600, method="thread")
+    def test_authorizer_raises_exception(self):
+        headers = {"header": "myheader"}
+        query_string = {"query": "myquery"}
+
+        response = requests.get(f"{self.url}/{self.endpoint}", headers=headers, params=query_string, timeout=300)
+        response_json = response.json()
+
+        self.assertEqual(response.status_code, 502)
+        self.assertEqual(response_json, {"message": "Internal server error"})
+
+
+class TestInvalidSwaggerTemplateUsingUnsupportedType(WritableStartApiIntegBaseClass):
+    """
+    Test using an invalid Lambda authorizer type
+    """
+
+    do_collect_cmd_init_output = True
+
+    template_content = """AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+
+Resources:
+  HttpApiOpenApi:
+    Type: AWS::ApiGatewayV2::Api
+    Properties:
+      Body:
+        openapi: "3.0"
+        info:
+          title: HttpApiOpenApi
+        components:
+          securitySchemes:
+            Authorizer:
+              type: apiKey
+              in: header
+              name: notused
+              "x-amazon-apigateway-authorizer":
+                authorizerPayloadFormatVersion: "2.0"
+                type: "bad type"
+                identitySource: "$request.header.header, $request.querystring.query"
+                authorizerUri: arn:aws:apigateway:us-east-1:lambda:path/2015-03-31/functions/arn:aws:lambda:us-east-1:123456789012:function:AuthorizerFunction/invocations
+"""
+
+    @pytest.mark.flaky(reruns=3)
+    @pytest.mark.timeout(timeout=10, method="thread")
+    def test_invalid_template(self):
+        self.assertIn(
+            "Lambda authorizer 'Authorizer' type 'bad type' is unsupported, skipping",
+            self.start_api_process_output,
+        )
+
+
+class TestInvalidSwaggerTemplateUsingSimpleResponseWithPayloadV1(WritableStartApiIntegBaseClass):
+    """
+    Test using simple response with wrong payload version
+    """
+
+    do_collect_cmd_init_output = True
+
+    template_content = """AWSTemplateFormatVersion: '2010-09-09'
```

Comment:
Just curious, why we didn't keep template contents in a yaml file but kept it here?

Reply:
Since these are incomplete + invalid templates that test a single case, I thought that it would be easier to read if it was just embedded in the code itself. No other reason aside from that. We could have them all as yaml files if we wanted to.